adam/curriculum_to_html.py

```diff
@@ -0,0 +1,184 @@
+import os
+from typing import List
+
+from attr import attrs
+
+from adam.experiment import InstanceGroup
+from adam.language.dependency import LinearizedDependencyTree
+from adam.perception import PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+)
+from adam.situation.high_level_semantics_situation import HighLevelSemanticsSituation
+
+
+@attrs(frozen=True, slots=True)
+class CurriculumToHtml:
+    """
+    Class to turn an `InstanceGroup` into an html document
+    """
+
+    @staticmethod
+    def generate(
+        instances: List[
+            InstanceGroup[
+                HighLevelSemanticsSituation,
+                LinearizedDependencyTree,
+                DevelopmentalPrimitivePerceptionFrame,
+            ]
+        ],
+        outputdestination: str,
+        title: str = "Instance Group",
+        overwrite: bool = False,
+    ):
+        r"""
+        Static method to take a list of `InstanceGroup`\ s and turns each one into an indivual page
+        
+        Given a list of InstanceGroups and an output directory of *outputdestination* along with 
+        a *title* for the pages the generator loops through each group and calls the internal 
+        method to create HTML pages. *overwrite* indicates is previously existing HTML files should
```

Comment:
In my experience, for things which are inexpensive to reproduce, it is less error prone to just nuke the output directory's current contents.